pyuvdata/uvdata/tests/test_uvdata.py

```diff
@@ -6792,3 +6793,113 @@ def test_multifile_read_errors(read_func, filelist):
         "Reading multiple files from class specific read functions is no "
         "longer supported."
     )
+
+
+def test_multifile_read_check():
+    """Test setting check_file_status=True when reading in files"""
+
+    uv = UVData()
+    uvh5_file = os.path.join(DATA_PATH, "zen.2458661.23480.HH.uvh5")
+
+    # Create a test file and remove header info to 'corrupt' it
+    testfile = os.path.join(DATA_PATH, "test", "zen.2458661.23480.HH.uvh5")
+    uv.read(uvh5_file)
+    uv.write_uvh5(testfile)
+    with h5py.File(testfile, "r+") as h5f:
+        del h5f["Header/ant_1_array"]
+
+    # Test that the expected error arises
+    with pytest.raises(KeyError) as cm:
+        uv.read(testfile, check_file_status=True)
+    assert "Unable to open object (object 'ant_1_array' doesn't exist)" in str(cm.value)
+
+    # Test when the corrupted file is at the end
+    uvTrue = UVData()
+    uvTrue.read(uvh5_file)
+    fileList = [uvh5_file, testfile]
+    with pytest.warns(UserWarning) as cm:
```

Comment:
If you add `match="your expected error message"` to the `pytest.warns` call it will check that the correct warning message was issued.